## importing req and known algorithms and models to see accuracy scores

In [1]:
import numpy as np
from sklearn.linear_model import LinearRegression
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.tree import DecisionTreeClassifier

## dataset

In [2]:
from sklearn.datasets import load_digits
digits = load_digits()

## import cross_val_score model 

In [3]:
from sklearn.model_selection import cross_val_score

## linear regression

In [4]:
li_scores = cross_val_score(LinearRegression(),digits.data, digits.target)
li_scores

array([0.48346048, 0.5583603 , 0.57534522, 0.5056632 , 0.40995457])

In [5]:
np.average(li_scores)

0.5065567550630685

## logistic regression

In [6]:
lg_scores = cross_val_score(LogisticRegression(solver = 'liblinear',max_iter=1000),digits.data, digits.target)
lg_scores

array([0.92222222, 0.88333333, 0.95264624, 0.95821727, 0.89415042])

In [7]:
np.average(lg_scores)

0.9221138966264315

## svm algorithm

In [8]:
svm_scores = cross_val_score(SVC(), digits.data, digits.target)
svm_scores

array([0.96111111, 0.94444444, 0.98328691, 0.98885794, 0.93871866])

In [9]:
np.average(svm_scores)

0.9632838130609718

## random forest classifier algorithm

In [10]:
rf_scores = cross_val_score(RandomForestClassifier(n_estimators=40),digits.data,digits.target)
rf_scores

array([0.92777778, 0.91111111, 0.94986072, 0.95264624, 0.9275766 ])

In [11]:
np.average(rf_scores)

0.9337944908696999

## decision tree classifier algorithm

In [12]:
de_scores = cross_val_score(DecisionTreeClassifier(), digits.data, digits.target)
de_scores

array([0.76666667, 0.71944444, 0.7994429 , 0.81337047, 0.79665738])

In [13]:
np.average(de_scores)

0.7791163726400495

### so far SVM performs best in predicting digits : 0.9632838130609718